cedar-policy-generators/src/schema.rs

```diff
@@ -820,49 +816,34 @@ impl Schema {
                 selected_indices.push(indices.swap_remove(index));
             }
 
-            Result::Ok(Some(
+            Result::Ok(
                 selected_indices
                     .iter()
                     .map(|&i| {
                         let (name, _) = &entity_types[i];
-                        tys.insert(build_qualified_entity_type_name(
-                            namespace.as_ref(),
-                            &name.clone().into(),
-                        ));
-                        name.clone().into()
+                        let etyp: ast::EntityType = ast::Name::from(name.clone()).into();
+                        tys.insert(build_qualified_entity_type_name(namespace.as_ref(), &etyp));
+                        etyp
                     })
-                    .collect::<Vec<Name>>(),
-            ))
+                    .collect::<Vec<ast::EntityType>>(),
+            )
         };
-        let mut principal_and_resource_types_exist = false;
         let mut actions: Vec<(SmolStr, ActionType)> = action_names
             .iter()
             .map(|name| {
                 Ok((
                     name.clone(),
                     ActionType {
                         applies_to: {
-                            let mut picked_resource_types =
-                                pick_entity_types(&mut resource_types, u)?;
-                            let mut picked_principal_types =
+                            let picked_resource_types = pick_entity_types(&mut resource_types, u)?;
+                            let picked_principal_types =
                                 pick_entity_types(&mut principal_types, u)?;
                             // If we already have resource_types and principal_types, randomly make them empty
-                            if principal_and_resource_types_exist {
-                                if u.ratio(1, 8)? {
-                                    picked_resource_types = None;
-                                }
-                                if u.ratio(1, 8)? {
-                                    picked_principal_types = None;
-                                }
-                            } else {
-                                principal_and_resource_types_exist = true;
-                            }
```

Comment:
yes I think we still need this

Reply:
Fixed it so the generator will sometimes make lists of length 0

Reply:
Actually, I explicitly made it so that the generator will create lists of minimum 1, and then later on decide whether to empty the list. I think we shouldn't initially generate lists of 0, since that's what caused the validity issue in the first place (no resource / principal types in all the actions). 

Would it be possible keep the minimum 1, and the subsequent coin flip to empty `picked_resource_types` and `picked_principal_types`

Reply:
Yeah sure

Reply:
thanks! I think you should also still keep the `principal_and_resource_types_exist` check so that it only deletes if we have at least one valid action.